src/odemis/acq/stream/_sync.py

```diff
@@ -1054,6 +1006,193 @@ def _runAcquisition(self, future):
+                    # So we log the memory usage here too.
+                    # TODO: Support also for Windows
+                    import odemis.util.driver as udriver
+                    memu = udriver.readMemoryUsage()
+                    # Too bad, need to use VmSize to get any good value
+                    logging.warning("Acquisition of repetition stream for "
+                                    "pixel %s timed out after %g s. "
+                                    "Memory usage is %d. Will try again",
+                                    px_idx, rep_time * 3 + 5, memu)
+                else:  # too fast to be possible (< the expected time - 5%)
+                    logging.warning("Repetition stream acquisition took less than %g s: %g s, will try again",
+                                    rep_time, dur)
+                failures += 1
+                if failures >= 3:
+                    # In three failures we just give up
+                    raise IOError("Repetition stream acquisition repeatedly fails to synchronize")
+                else:
+                    for s, sub, ad in zip(self._streams, self._subscribers, self._acq_data):
+                        s._dataflow.unsubscribe(sub)
+                        # Ensure we don't keep the data for this run
+                        ad[:] = ad[:n]
+
+                    # Restart the acquisition, hoping this time we will synchronize
+                    # properly
+                    time.sleep(1)
+                    self._ccd_df.subscribe(self._subscribers[-1])
+                    continue
+
+            # Normally, the SEM acquisitions have already completed
+            # get image for SEM streams (at least one for ebeam)
+            for s, sub, ce in zip(self._streams[:-1], self._subscribers[:-1], self._acq_complete[:-1]):
+                if not ce.wait(sem_time * 1.5 + 5):
+                    raise TimeoutError("Acquisition of SEM pixel %s timed out after %g s"
+                                       % (px_idx, sem_time * 1.5 + 5))
+                logging.debug("Got synchronisation from %s", s)
+                s._dataflow.unsubscribe(sub)
+
+            if self._acq_state == CANCELLED:
+                raise CancelledError()
+
+            # MD_POS default to the center of the stage, but it needs to be
+            # the position of the e-beam (without the shift for drift correction)
+            raw_pos = self._acq_data[0][-1].metadata[MD_POS]
+            drift_shift = self._dc_estimator.tot_drift if self._dc_estimator else (0, 0)
+            cor_pos = (raw_pos[0] + drift_shift[0] * sub_pxs[0],
+                       raw_pos[1] - drift_shift[1] * sub_pxs[1])  # Y is upside down
+            ccd_data = self._acq_data[-1][-1]
+            ccd_data.metadata[MD_POS] = cor_pos
+
+            self._acq_data[-1][-1] = self._preprocessData(len(self._streams) - 1, ccd_data, px_idx)
+            logging.debug("Processed CCD data %d = %s", n, px_idx)
+
+            n += 1
```

Comment:
As n is an int, it will only be updated here and not in the caller. You could just put a `+ 1` in the two lines using it, to clarify this.